pom.xml

```diff
@@ -600,5 +630,13 @@ file comparators, endian transformation classes, and much more.
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>release</id>
+      <properties>
+        <!-- Enable Moditect for releases. This requires JDK 9+ for the build -->
```

Comment:
That's not good. I build releases with Java 8 since we want to bind to the JRE 8 APIs and not 9 especially with problems due to the NIO Buffer class changes.

Reply:
What are the issues with NIO Buffer classes if you build with Java 9 and --release=8 ?

Reply:
@garydgregory I think, that **release building** with Java 9+ is a more than reasonable requirement, if we want to fit into the Java 9 ecosphere. (There is no question, that we want that, isn't there?)

@martin-g I'd like to see two improvements:

1. Move the plugin definition into a profile, that enables the moditect-maven-plugin only when building with Java 9+. Alternatively, set the property moditect.skip for building with Java 8.
2. Make sure, that the apache-release profile enforces building with Java 9+.

Regarding commons-parent: My personal stance is, that we have covered commons-io, and commons-fileupload now (Perhaps, cover a few others, in case of demand.) Let's wait, and see, until we did a release with that, then approach parent.

Reply:
@jochenw Done with https://github.com/apache/commons-io/pull/268/commits/37d5e7d09e0899aa15dd69a20d976d3a31bf17d9